### ISSUE TITLE
TypeScript defintions

### ISSUE THREAD

**Issue by CXuesong** [JWally/jsLPSolver#95]
First of all, thank you for bringing this awesome LP/MIP solver into JavaScript world! I've tried your solver to solve some simple MIP problems and it seems promising.

However, up till now there seems no available TypeScript defintion for this package. And IMO a fine TS defintion can provide us TS users more information about the features available in your library. While playing with this package, I've written a simple (and rather in-complete) `.d.ts` file that, hopefully, may be added into your repo and published with js files. Now I'm seeking your help to complete it.

Basically, I'd like to create a fork open a PR in your repo for this, and (slowly but) gradually add more and more members in the `.d.ts` file. You may provide insights about APIs yet to be added in the PR during the whole process. Does this look okay to you?

p.s. Please expect this process to be rather slow. I'm [working on a project](https://github.com/crystal-pool/warriors-family-tree) that may involve solving some MIP problems during my free time.
